ftw/news/tests/test_news_listing.py

```diff
@@ -0,0 +1,144 @@
+from datetime import timedelta, datetime
+from DateTime.DateTime import DateTime
+from ftw.builder import Builder
+from ftw.builder import create
+from ftw.news.testing import FTW_NEWS_FUNCTIONAL_TESTING
+from ftw.testbrowser import browsing
+from ftw.testbrowser.pages import plone
+from plone.app.testing import setRoles
+from plone.app.testing import TEST_USER_ID
+from Products.CMFCore.utils import getToolByName
+from unittest2 import TestCase
+from ftw.builder.user import DEFAULT_PASSWORD
+
+
+def set_allow_anonymous_view_about(context, enable):
+    site_props = getToolByName(
+        context, 'portal_properties').site_properties
+    site_props.allowAnonymousViewAbout = enable
+
+
+class TestNewsListing(TestCase):
+
+    layer = FTW_NEWS_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer['portal']
+        setRoles(self.portal, TEST_USER_ID, ['Manager'])
+
+        self.news_folder = create(Builder('news folder')
+                                  .titled(u'A News Folder'))
+
+        yesterday = datetime.today() - timedelta(days=1)
+        self.news1 = create(Builder('news').titled(u'News Entry 1')
+                            .within(self.news_folder)
+                            .having(effective=yesterday)
+                            )
+
+        tomorrow = datetime.today() + timedelta(days=1)
+        self.news2 = create(Builder('news').titled(u'News Entry 2')
+                            .within(self.news_folder)
+                            .having(effective=tomorrow)
+                            )
+
+        set_allow_anonymous_view_about(self.news_folder, True)
+
+        self.portal.manage_permission('Access inactive portal content',
+                                      ['Contributor', 'Manager'],
+                                      acquire=False)
+
+        # Create user with local contributor role
+        self.contributor = create(Builder('user')
+                                  .named('Hugo', 'Boss')
+                                  .with_roles('Contributor'))
+
+        # Create user without special roles
+        self.member = create(Builder('user')
+                             .named('John', 'Doe')
+                             .with_roles('Member'))
+
+    def _is_author_visible(self, browser):
+        browser.login().visit(self.news_folder)
+        return '<span class="documentAuthor"' in browser.contents
+
+    @browsing
+    def test_member_sees_author_when_aava_disabled(self, browser):
+        set_allow_anonymous_view_about(self.news_folder, True)
+        browser.login(username=self.member.getProperty('id'),
+                      password=DEFAULT_PASSWORD).open(self.news_folder)
```

Comment:
login can be done easier, since ftw.testbrowser work swell with ftw.builder's user builder:
```python
browser.login(self.member).open(self.news_folder)
```
:sunglasses: